.changelog/371.txt

```diff
@@ -0,0 +1,3 @@
+```release-note:bug
+resource/ec_deployment: Fix a bug affecting partial version upgrades, where some resources upgraded but others did not, leaving the `version` argument set to the configured value as if the upgrade was successful. The upgrade could not be retried and it was unable to detect drift.
```

Comment:
I'm not so sure on the wording here, we might need a wordsmith to come in and fix this strange sentence. @alaudazzi if you have some minutes 🙂